Submitty/Submitty
# [Router] New URL for CourseMaterialsController

### URL Changes

**Old:** 
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=view_course_materials_page`
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=delete_course_material_file`
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=delete_course_material_folder`
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=download_course_material_zip`
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=modify_course_materials_file_permission`
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=modify_course_materials_time_stamp`
- `/index.php?semester=s19&course=sample&component=grading&page=course_materials&action=upload_course_materials_files`

**New:** 
- `/s19/development/course_materials`
- `/s19/development/course_materials/delete`
- `/s19/development/course_materials/delete_folder`
- `/s19/development/course_materials/download_zip`
- `/s19/development/course_materials/modify_permission`
- `/s19/development/course_materials/modify_timestamp`
- `/s19/development/course_materials/upload`

### Tested Entrances
As instructor:
- Sidebar -> Course Materials
- Upload, delete file, delete folder, download zip, modify permission & timestamp.

As student:
- Sidebar -> Course Materials
- Download file, check correctness of permission.

### Other Information

I pushed this and found #3923 is also about course materials, but luckily we are not editing the same files. Just several lines' conflicts.
